src/main/scala/com/lucidchart/open/relate/StatementPreparer.scala

```diff
@@ -61,6 +61,12 @@ private[relate] sealed trait StatementPreparer {
 private[relate] trait BaseStatementPreparer extends StatementPreparer {
   protected def applyParams(stmt: PreparedStatement)
   protected def parsedQuery: String
+  protected def timeout: Option[Int] = None
+
+  protected def setTimeout(stmt: PreparedStatement): Unit = for {
+    seconds <- timeout
+    stmt <- Option(stmt)
+  } yield (stmt.setQueryTimeout(seconds))
```

Comment:
What does this mean?

On Tue, Jun 16, 2015 at 8:20 PM Gregg Hernandez <notifications@github.com>
wrote:

> In src/main/scala/com/lucidchart/open/relate/StatementPreparer.scala
> <https://github.com/lucidsoftware/relate/pull/24#discussion_r32590749>:
>
> > @@ -61,6 +61,12 @@ private[relate] sealed trait StatementPreparer {
> >  private[relate] trait BaseStatementPreparer extends StatementPreparer {
> >    protected def applyParams(stmt: PreparedStatement)
> >    protected def parsedQuery: String
> > +  protected def timeout: Option[Int] = None
> > +
> > +  protected def setTimeout(stmt: PreparedStatement): Unit = for {
> > +    seconds <- timeout
> > +    stmt <- Option(stmt)
> > +  } yield (stmt.setQueryTimeout(seconds))
>
> It happened when I was testing on the user service.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lucidsoftware/relate/pull/24/files#r32590749>.
>